include/language-factory.php

```diff
@@ -144,4 +147,106 @@ private static function get_languages() {
 
 		return self::$languages;
 	}
+
+
+	/**
+	 * Creates flag_url and flag language properties. Also takes care of custom flag.
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $flag_code Flag code.
+	 * @param string $name      Language name.
+	 * @param string $slug      Language slug.
+	 * @param string $locale    Language locale.
+	 * @return array {
+	 *     Array of the flag properties.
+	 *     @type string  $flag_url        URL of the flag.
+	 *     @type string  $flag            HTML markup of the flag.
+	 *     @type string  $custom_flag_url Optional. URL of the custom flag if it exists.
+	 *     @type string  $custom_flag     Optional. HTML markup of the custom flag if it exists.
+	 * }
+	 *
+	 * @phpstan-return array{
+	 *     flag_url: non-empty-string
+	 *     flag: non-empty-string
+	 *     custom_flag_url?: non-empty-string
+	 *     custom_flag?: non-empty-string
+	 * }
+	 */
+	private static function get_flag( $flag_code, $name, $slug, $locale ) {
+		$flags = array( 'flag' => PLL_Language::get_flag_informations( $flag_code ) );
+
+		// Custom flags?
+		$directories = array(
+			PLL_LOCAL_DIR,
+			get_stylesheet_directory() . '/polylang',
+			get_template_directory() . '/polylang',
+		);
+
+		foreach ( $directories as $dir ) {
+			if ( file_exists( $file = "{$dir}/{$locale}.png" ) || file_exists( $file = "{$dir}/{$locale}.jpg" ) || file_exists( $file = "{$dir}/{$locale}.svg" ) ) {
+				$flags['custom_flag']['url'] = content_url( '/' . str_replace( WP_CONTENT_DIR, '', $file ) );
+				break;
+			}
+		}
+
+		/**
+		 * Filters the custom flag informations.
```

Comment:
```suggestion
		 * Filters the custom flag information.
```

Reply:
"information" is uncountable IMHO.

Reply:
Isn't that why we shouldn't add an "s"?

Reply:
Yup, for instance, we have to use "a piece of information" to about a specific one.

Reply:
Sorry, didn't read correctly...